integration/walletextension/wallet_extension_test.go

```diff
@@ -632,7 +631,25 @@ func createObscuroNetwork(t *testing.T) (func(), error) {
 	if err != nil {
 		return obscuroNetwork.TearDown, err
 	}
-	makeEthJSONReqAsJSON(t, walletExtensionAddr, walletextension.ReqJSONMethodSendRawTx, []interface{}{txBinaryHex})
+	sendTxJSON := makeEthJSONReqAsJSON(t, walletExtensionAddr, walletextension.ReqJSONMethodSendRawTx, []interface{}{txBinaryHex})
+
+	// We wait for the Obscuro ERC20 contract deployment to be deployed.
+	txHash, ok := sendTxJSON[walletextension.RespJSONKeyResult].(string)
+	if !ok {
+		panic("could not retrieve transaction hash from JSON result")
+	}
+	counter := 0
+	for {
+		if counter > 20 {
+			t.Fatalf("could not get receipt for Obscuro ERC20 deployment transaction after 20 seconds")
+		}
+		txReceiptResp := makeEthJSONReq(t, walletExtensionAddr, walletextension.ReqJSONMethodGetTxReceipt, []string{txHash})
+		if !strings.Contains(string(txReceiptResp), "could not retrieve transaction with hash") {
```

Comment:
The exiting of this loop is slightly hard to wrap your head around on the first read-through.

Maybe worth a comment to break it down for those of us running on little sleep, along the lines of: "Breakout if we find the hash, or if we see an unexpected failure. If just hash not found we continue to retry."

Reply:
Fixed.